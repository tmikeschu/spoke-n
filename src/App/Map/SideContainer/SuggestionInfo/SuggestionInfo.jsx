import React from 'react'
import PropTypes from 'prop-types'

const SuggestionInfo = props => {
  const { currentSuggestion, suggestionInfoIsActive } = props

  const categories = {
    "stay": "Place to stay",
    "checkout": "Cool spot",
    "avoid": "Avoid this place",
    "bike_shop": "Bike shop",
    "other": "Other",
  }

  return (
    <article className="suggestion-info">
      <h4>Suggestion Info</h4>
      <p style={{display: currentSuggestion ? "none" : "block"}}>(click an existing pin on the map)</p>
      <section className="text" style={{ display: suggestionInfoIsActive ? "initial" : "none" }}>
        <p><span>Label</span>: { currentSuggestion && currentSuggestion.label}</p>
        <p><span>Description</span>: { currentSuggestion && currentSuggestion.description}</p>
        <p><span>Category</span>: { currentSuggestion && categories[currentSuggestion.category]}</p>
      </section>
    </article>
  )
}

SuggestionInfo.propTypes = {
  currentSuggestion: PropTypes.object,
  suggestionInfoIsActive: PropTypes.bool.isRequired,
}

export default SuggestionInfo
