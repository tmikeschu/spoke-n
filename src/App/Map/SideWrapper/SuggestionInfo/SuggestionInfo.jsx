import React from 'react'
import PropTypes from 'prop-types'

const SuggestionInfo = ({ currentSuggestion, suggestionInfoIsActive }) => {
  const categories = {
    "stay": "Place to stay",
    "checkout": "Cool spot",
    "avoid": "Avoid this place",
    "bike_shop": "Bike shop",
    "other": "Other",
  }

  return (
    <article className="SuggestionInfo">
      <h4>Suggestion Info</h4>
      <p style={{display: suggestionInfoIsActive ? "none" : "block"}}>(click an existing pin on the map)</p>
      <section className="text" style={{ display: suggestionInfoIsActive ? "initial" : "none" }}>
        <p><span>Label</span>: { currentSuggestion.label }</p>
        <p><span>Description</span>: { currentSuggestion.description }</p>
        <p><span>Category</span>: {  categories[currentSuggestion.category] }</p>
      </section>
    </article>
  )
}

SuggestionInfo.propTypes = {
  currentSuggestion: PropTypes.object.isRequired,
  suggestionInfoIsActive: PropTypes.bool.isRequired,
}

export default SuggestionInfo

