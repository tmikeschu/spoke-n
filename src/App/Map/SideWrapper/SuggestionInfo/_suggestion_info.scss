.suggestion-info {
  border: 1px dotted $yellow;
  display: flex;
  margin-top: 5%;
  flex-direction: column;
  h4, p, section {
    margin: 2% 1%;
  }
  .text {
    width: 100%;
    p {
      margin: 1% 0;
      span {
        font-weight: bold;
        color: $brown;
      }
    }
  }
  @media (max-width: 640px) {
    font-size: .5em;
  }
}
