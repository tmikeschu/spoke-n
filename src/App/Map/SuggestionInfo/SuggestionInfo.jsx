import React, { Component } from 'react';
import '../../App.css'

export default class SuggestionInfo extends Component {
  render() {
    const categories = {
      "stay": "Place to stay",
      "checkout": "Cool spot",
      "avoid": "Avoid this place",
      "bike_shop": "Bike shop",
      "other": "Other",
    }

    return (
      <article className="suggestion-info">
        <h4>Suggestion Info</h4>
        <p style={{display: this.props.currentSuggestion ? "none" : "block"}}>(click an existing pin on the map)</p>
        <section className="text" style={{ display: this.props.suggestionInfoIsActive ? "initial" : "none" }}>
          <p><span>Label</span>: { this.props.currentSuggestion && this.props.currentSuggestion.label}</p>
          <p><span>Description</span>: { this.props.currentSuggestion && this.props.currentSuggestion.description}</p>
          <p><span>Category</span>: { this.props.currentSuggestion && categories[this.props.currentSuggestion.category]}</p>
        </section>
      </article>
    );
  }
}
