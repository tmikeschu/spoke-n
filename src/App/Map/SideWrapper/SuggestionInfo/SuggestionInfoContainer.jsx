import { connect } from "react-redux"
import SuggestionInfo from "./SuggestionInfo"

const mapStateToProps = ({ map }) => ({
  currentSuggestion: map.currentSuggestion,
  suggestionInfoIsActive: map.suggestionInfoIsActive
})

export default connect(mapStateToProps)(SuggestionInfo)

