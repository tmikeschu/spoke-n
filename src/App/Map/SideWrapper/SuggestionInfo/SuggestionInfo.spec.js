import React from "react"
import { shallow } from "enzyme"
import SuggestionInfo from "./SuggestionInfo"

describe("<SuggestionInfo />", () => {
  const props = {
    currentSuggestion: {},
    suggestionInfoIsActive: false
  }
  const wrapper = shallow(<SuggestionInfo {...props} />)

  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(wrapper.getNodes()).toMatchSnapshot()
    })
  })
})
