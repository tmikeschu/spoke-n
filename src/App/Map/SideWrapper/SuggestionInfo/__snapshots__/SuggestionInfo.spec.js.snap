// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuggestionInfo /> snapshot is valid 1`] = `
Array [
  <article
    className="suggestion-info"
>
    <h4>
        Suggestion Info
    </h4>
    <p
        style={
            Object {
                "display": "block",
              }
        }
    >
        (click an existing pin on the map)
    </p>
    <section
        className="text"
        style={
            Object {
                "display": "none",
              }
        }
    >
        <p>
            <span>
                Label
            </span>
            : 
        </p>
        <p>
            <span>
                Description
            </span>
            : 
        </p>
        <p>
            <span>
                Category
            </span>
            : 
        </p>
    </section>
</article>,
]
`;
